### ISSUE TITLE
metainfo causes appstream-util segfault

### ISSUE THREAD

**Issue by limburgher** [frescobaldi/frescobaldi#1296]
I'd like to include the included org.frescobaldi.Frescobaldi.metainfo.xml in the Fedora rpm, but I have to run appstream-util validate-relax --nonet on it at build time, and something about this file causes a segfault on all versions of Fedora.

**Comment by fedelibre** [comment]
I see that the segfault occurs only with the template file and not with the generated file:

```
$ appstream-util validate-relax --nonet org.frescobaldi.Frescobaldi.metainfo.xml
org.frescobaldi.Frescobaldi.metainfo.xml: OK

$ appstream-util validate-relax --nonet org.frescobaldi.Frescobaldi.metainfo.xml.in
org.frescobaldi.Frescobaldi.metainfo.xml.in: Errore di segmentazione (core dump creato)
```

I'll try to understand why. Maybe some XML error I cannot see.

Anyway, I think you should validate the final file and not the template file.
You can generate the final metainfo file first (`make -C linux`) and then validate the generated file.

**Comment by fedelibre** [comment]
Ah.
Yesterday I got the Ok for the generated file (see above), but today I'm getting the segfault. Strange...

`appstreamcli validate` does a better job in finding the error:

```
$ appstreamcli validate org.frescobaldi.Frescobaldi.metainfo.xml.in 
I: org.frescobaldi.Frescobaldi:63: url-not-secure http://frescobaldi.org/
E: org.frescobaldi.Frescobaldi:73: description-enum-item-invalid ul
```

There's a nested bullet
[0001-Remove-nested-list-from-metainfo-release.patch.txt](https://github.com/frescobaldi/frescobaldi/files/4561375/0001-Remove-nested-list-from-metainfo-release.patch.txt)
 list in 3.1.2 release. It's not allowed by the spec.
This will be solved in master as soon as #1289 is merged.

As I don't think the next release 3.2 will be released soon, can you apply the attached patch to 3.1.2 package?
Thanks

**Comment by limburgher** [comment]
Will do, thank you!
